### PR TITLE
[idlharness.js] Handle callbacks in very basic manner

### LINKED ISSUE #14254: Intermittent "Unrecognized type EventHandlerNonNull" in html/dom/interfaces.https.html
I've seen this in https://github.com/web-platform-tests/wpt/pull/14252, but also before that. I tried to reproduce at the time, but failed.

### PR DESCRIPTION
**Pull request by lukebjerring**
Should fix https://github.com/web-platform-tests/wpt/issues/14254 (since the handlers being attached are actually the right type).

Doesn't do anything clever about the arguments/signature for the callback just yet.

### CODE REVIEW

File: html/dom/interfaces.https.html [hunk 1]
```diff
@@ -48,7 +56,7 @@ <h1>HTML IDL tests</h1>
       HTMLOptionsCollection: ['document.createElement("select").options'],
       DOMStringMap: ['document.head.dataset'],
       Transferable: [],
-      Document: ['iframe.contentDocument', 'new Document()'],
+      Document: ['iframe.contentDocument', 'documentWithHandlers'],
```
Comment (bzbarsky) [inline html/dom/interfaces.https.html]: Why not keep both `new Document()` and `documentWithHandlers` in the list here?

File: resources/idlharness.js [hunk 2]
```diff
@@ -1163,9 +1162,14 @@ IdlArray.prototype.assert_type_is = function(value, type)
     {
         // TODO: Test when we actually have something to test this on
     }
+    else if (this.members[type] instanceof IdlCallback)
+    {
+        assert_equals(typeof value, "function");
+        // TODO: Better assertions about signature?
```
Comment (bzbarsky) [inline resources/idlharness.js]: You can't assert anything about the "signature" of a JS function.

**Review by bzbarsky — COMMENTED** [review]
Thank you for fixing this!

**Review by bzbarsky — APPROVED** [review]

**Comment by lukebjerring** [comment]
Stability job timed out; results indicate this doesn't make things regress, so merging.

**Pull request merged**